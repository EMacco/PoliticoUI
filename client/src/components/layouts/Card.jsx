import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CardText from '../common/CardText';
import CardLink from '../common/CardLink';

const Card = ({ name, image, content, hidden }) => {
  const cardContent = content.map(details => {
    const {
      type,
      title,
      value,
      officeId,
      officeName,
      numberOfCandidates,
      hidden,
      text,
      style
    } = details;
    let design;

    if (hidden === 'true') return;
    switch (type) {
      case 'link':
        design = (
          <CardLink
            officeId={officeId}
            officeName={officeName}
            numberOfCandidates={numberOfCandidates}
            key={Math.random().toString()}
          />
        );
        break;
      case 'button':
        design = (
          <button className={style} key={Math.random().toString()}>
            {text}
          </button>
        );
        break;
      case 'span':
        design = (
          <span className={style} key={Math.random().toString()}>
            {text}
          </span>
        );
        break;
      default:
        design = <CardText title={title} value={value} key={Math.random().toString()} />;
    }

    return design;
  });

  return (
    <Fragment>
      <div
        className={
          hidden === 'true'
            ? 'individual-person-container hidden-div'
            : 'individual-person-container'
        }
      >
        <div>
          <img src={image} />
        </div>
        <div className="profile-description-text">
          <label>
            <span className="profile-answers">{name.charAt(0).toUpperCase() + name.substr(1)}</span>
          </label>

          {cardContent}
        </div>
      </div>
    </Fragment>
  );
};

Card.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  content: PropTypes.array,
  hidden: PropTypes.string
};

Card.defaultProps = {
  name: '',
  image: '',
  content: [],
  hidden: 'false'
};

export default Card;
