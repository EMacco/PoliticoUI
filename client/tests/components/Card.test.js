import React from 'react';
import Card from '@components/layouts/Card';

const props = {
  name: 'emma',
  image: '',
  content: [
    {
      type: 'link',
      title: 'test',
      value: 'This is the value',
      officeId: 1,
      officeName: 'Office name',
      numberOfCandidates: 4,
      hidden: false,
      text: 'Hahaha',
      style: '',
      onClick: () => {},
      id: 1,
      key: 1
    },
    {
      type: 'button',
      title: 'test',
      value: 'This is the value',
      officeId: 1,
      officeName: 'Office name',
      numberOfCandidates: 4,
      hidden: false,
      text: 'Hahaha',
      style: '',
      onClick: () => {},
      id: 1,
      key: 2
    },
    {
      type: 'span',
      title: 'test',
      value: 'This is the value',
      officeId: 1,
      officeName: 'Office name',
      numberOfCandidates: 4,
      hidden: false,
      text: 'Hahaha',
      style: '',
      onClick: () => {},
      id: 1,
      key: 3
    },
    {
      title: 'test',
      value: 'This is the value',
      officeId: 1,
      officeName: 'Office name',
      numberOfCandidates: 4,
      hidden: false,
      text: 'Hahaha',
      style: '',
      onClick: () => {},
      id: 1,
      key: 4
    },
    {
      type: 'empty',
      title: 'test',
      value: 'This is the value',
      officeId: 1,
      officeName: 'Office name',
      numberOfCandidates: 4,
      hidden: false,
      text: 'Hahaha',
      style: '',
      onClick: () => {},
      id: 1
    }
  ],
  hidden: 'false'
};

const setup = () => {
  const wrapper = shallow(<Card {...props} />);
  return wrapper;
};

describe('<Card />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the Card component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Card')).toBeTruthy();
  });
});
