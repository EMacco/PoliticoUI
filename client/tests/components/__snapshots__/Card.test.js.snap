// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the Card component correctly 1`] = `
<Fragment>
  <div
    className="individual-person-container"
  >
    <div>
      <img
        src=""
      />
    </div>
    <div
      className="profile-description-text"
    >
      <label>
        <span
          className="profile-answers"
        >
          Emma
        </span>
      </label>
      <CardLink
        key="1"
        numberOfCandidates={4}
        officeId={1}
        officeName="Office name"
      />
      <button
        className=""
        id={1}
        key="2"
        onClick={[Function]}
      >
        Hahaha
      </button>
      <span
        className=""
        key="3"
      >
        Hahaha
      </span>
      <CardText
        key="4"
        title="test"
        value="This is the value"
      />
    </div>
  </div>
</Fragment>
`;
